### ISSUE TITLE
[Firefox on Android] blocking (off) when opening new tabs in Firefox from 3rd party apps

### ISSUE THREAD

**Issue by Snow8** [uBlock-LLC/uBlock#1017]
Hi,

first off all I really appreciate the hard work from the devs that worked on uBlock for Firefox on Android.

I have the following issue. Whenever I open links from 3rd party apps (in my case Press, Reddit, Twitter, and a few others) in Firefox, ublock doesnt block anything until I reload the page manually after the tab was opened. Before reloading and ublock working as intended, it says in the "menu tab" --> ublock(off).

I am running Firefox Stable on Android 5.0.1 on a Nexus 5. All other apps are up to date as well for what it is worth.

My ublock settings are:

"I am an advanced user" enabled
- \* 3p block (disabled on a per site setting when css is broken)
- \* 3p-frame block
  3rd Party Filters are pretty much standard with the German List + Social blocking enabled.

If you need any further infos please let me know.

**Comment by AlexVallat** [comment]
Confirmed. I'm investigating.

**Comment by AlexVallat** [comment]
OK, some analysis here. Firstly, when Fennec is opening the new tab from the 3rd party intent you get the same effect as in #1001 where the root document is treated as being behind the scenes. I applied the same test fix for a better tab ID, but there was then a further issue that the `onTabSelect` handler was being called for the new tab after it had started to load, but before the `.currentURI` had been updated from `about:blank`. This resulted in the pageStore for that tab being rebound to `about:blank`. To resolve this, I added a check to see if the page is currently loading before calling `onNavigation` for `about:blank` (or `about:newtab`).

Changes in https://github.com/gorhill/uBlock/pull/1021
